src/dmell_line.c

```diff
@@ -253,33 +253,31 @@ int dmell_run_line(const char* line, size_t len)
     int last_exit_code = 0;
     int result = 0;
     bool first_command = true;
+    dmell_line_sep_t prev_sep = dmell_line_sep_none;
     while( ptr < end_ptr && *ptr != '\0' )
     {
         // Find the next command separator
         dmell_line_sep_t sep = dmell_line_sep_none;
         const char* sep_ptr = find_next_command_separator( ptr, end_ptr, &sep );
 
         // Check if we should execute the current command - lazy evaluation
-        if( first_command || should_execute_command( last_exit_code, sep ) )
+        // Use the previous separator to decide if the current command should run
+        if( first_command || should_execute_command( last_exit_code, prev_sep ) )
```

Comment:
Wydaje mi się, że ta flaga first_command już nie jest potrzebna - czy możesz to sprawdzić?